lib/grammers-client/src/client/chats.rs

```diff
@@ -725,6 +725,124 @@ impl Client {
         let permissions = ParticipantPermissions::Channel(participant.participant);
         Ok(permissions)
     }
+    
+    #[cfg(feature = "parse_invite_link")]
+    fn parse_invite_link(invite_link: &str) -> Option<String> {
+        let url_parse_result = url::Url::parse(invite_link);
+        if url_parse_result.is_err() {
+            return None;
+        }
+
+        let url_parse = url_parse_result.unwrap();
+        let scheme = url_parse.scheme();
+        let path = url_parse.path();
+        if url_parse.host_str().is_none() || !vec!["https", "http"].contains(&scheme) {
+            return None;
+        }
+        let host = url_parse.host_str().unwrap();
+        let hosts = vec![
+            "t.me",
+            "telegram.me",
+            "telegram.dog",
+            "tg.dev",
+            "telegram.me",
+            "telesco.pe",
+        ];
+
+        if !hosts.contains(&host) {
+            return None;
+        }
+        let paths = path.split("/").collect::<Vec<&str>>();
+
+        if paths.len() == 1 {
+            if paths[0].starts_with("+") {
+                return Some(paths[0].replace("+", ""));
+            }
+            return None;
+        }
+
+        if paths.len() > 1 {
+            if paths[0].starts_with("joinchat") {
+                return Some(paths[1].to_string());
+            }
+            if paths[0].starts_with("+") {
+                return Some(paths[0].replace("+", ""));
+            }
+            return None;
+        }
+
+        None
+    }
+
+    /// Accept an invite link to join the corresponding private chat.
+    ///
+    /// If the chat is public (has a public username), [`Client::join_chat`](Client::join_chat) should be used instead.
+    pub async fn accept_invite_link(
+        &mut self,
+        invite_link: &str,
+    ) -> Result<tl::enums::Updates, InvocationError> {
+        #[cfg(not(feature = "parse_invite_link"))]
+        let hash = invite_link.to_string();
+        #[cfg(feature = "parse_invite_link")]
+        let hash = {
```

Comment:
This is rather confusing. A crate feature should generally not change the expected format of input parameters.

Imagine I depend on crate `foo` which depends on `grammers-client`. `foo` doesn't have the feature enabled, so it uses the hash. Now, I also depend on `grammers-client`, but enable the feature. Now when I use `foo`, it won't work correctly, because the feature is enabled.

I think we should get rid of the feature and make the `url` dependency mandatory (so it will always be available).

Reply:
To clarify what I meant in https://github.com/Lonami/grammers/pull/151#discussion_r992472784: we should pick one option only. I think it's fine to always have `url` available and accept all forms of invite links (option 2 only).

Reply:
So do I need to modify parse_invite_link to become the url?

Reply:
I removed the parameter to pass the hash directly, and updated it to accept_invite_link only when feature = "parse_invite_link" is enabled. Is this normal?

Reply:
Not what I had in mind but that'll work. We can change it later.